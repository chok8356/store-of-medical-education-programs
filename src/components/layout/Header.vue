<template>
  <nav class="navbar navbar-dark bg-dark box-shadow">
    <div class="container d-flex justify-content-between">
      <a
        class="navbar-brand d-flex align-items-center"
        href="/">Магазин</a>
      <b-button
        v-if="!inCart"
        @click="onCartClick">
        В корзине: {{ cartQuantity }} ({{ cartTotal | currency }})
      </b-button>
    </div>
  </nav>
</template>

<script lang="ts">
import { BButton } from 'bootstrap-vue'
import { Vue, Component, Prop } from 'vue-property-decorator'
import { currency } from '@/filters'

@Component({
  name: 'Header',
  filters: {
    currency
  },
  components: {
    BButton
  }
})
export default class Header extends Vue {
  @Prop({ type: String, default: '0.00' }) cartTotal!: number
  @Prop({ type: Number, default: 0 }) cartQuantity!: number

  get inCart() {
    return this.$route && this.$route.name === 'Cart'
  }

  onCartClick() {
    this.$router.push({ name: 'Cart' })
  }
}
</script>
