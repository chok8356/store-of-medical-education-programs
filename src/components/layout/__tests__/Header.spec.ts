import { shallowMount } from '@vue/test-utils'
import { BButton } from 'bootstrap-vue'

import Header from '../Header.vue'

describe('Header', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Header)
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('render correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('exist default props', () => {
    expect(wrapper.props('cartTotal')).toBe('0.00')
    expect(wrapper.props('cartQuantity')).toBe(0)
  })

  it('in cart', () => {
    wrapper = shallowMount(Header, {
      mocks: {
        $route: {
          name: 'Cart'
        }
      }
    })
    expect(wrapper.vm.inCart).toBe(true)
  })

  it('click on cart change router', async() => {
    wrapper = shallowMount(Header, {
      mocks: {
        $router: { push: () => {} }
      }
    })
    const routerPush = jest.spyOn(wrapper.vm.$router, 'push')
    const buttonComponent = wrapper.findComponent(BButton)
    await buttonComponent.trigger('click')
    expect(routerPush).toBeCalled()
  })
})
