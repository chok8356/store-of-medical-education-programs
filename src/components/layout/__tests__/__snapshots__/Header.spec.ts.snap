// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<nav
  class="navbar navbar-dark bg-dark box-shadow"
>
  <div
    class="container d-flex justify-content-between"
  >
    <a
      class="navbar-brand d-flex align-items-center"
      href="/"
    >
      Магазин
    </a>
     
    <b-button-stub
      tag="button"
      target="_self"
      type="button"
      variant="secondary"
    >
      
      В корзине: 0 (0,00 ₽)
    
    </b-button-stub>
  </div>
</nav>
`;
